src/osmviz/manager.py

```diff
@@ -257,8 +293,11 @@ def getTileURL(self, tile_coord, zoom):
         Given x, y coord of the tile to download, and the zoom level,
         returns the URL from which to download the image.
         """
-        params = (zoom, tile_coord[0], tile_coord[1])
-        return self.server+"/%d/%d/%d.png" % params
+        return self.url \
+            .replace("{z}", str(zoom)) \
+            .replace("{x}", str(tile_coord[0])) \
+            .replace("{y}", str(tile_coord[1])) \
+            .replace("{s}", str(self.scale))
```

Comment:
Hmm there's one problem with this approach: trying to replace placeholders that don't exist (such as the optional "{s}") will result in a KeyError. The "replace" solution doesn't have this drawback. It is possible to prevent such errors by using a custom formatter (see e.g. http://stackoverflow.com/questions/3536303/python-string-format-suppress-silent-keyerror-indexerror) but that doesn't make for a very elegant solution, also taking into account the extra arguments we'd have to feed each format call.